pkg/domain/infra/abi/play.go

```diff
@@ -114,6 +117,26 @@ func (ic *ContainerEngine) PlayKube(ctx context.Context, body io.Reader, options
 	report := &entities.PlayKubeReport{}
 	validKinds := 0
 
+	// when no network options are specified, create a common network for all the pods
+	if len(options.Networks) == 0 {
+		defaultNetworkExists, err := ic.NetworkExists(ctx, kubeDefaultNetwork)
```

Comment:
please move this block into the `if podOpt.Net.Network.NSMode != "host" && len(options.Networks) == 0 {` branch below, no need to create this when host is used

Reply:
If left here it will check the network existence only once per kube invocation, when I'll move it inside the playKubePod it will check for network existence whenever a pod is created. The trade-off having the network created even when it is not in use is minimal - eventually it will get created anyway.
I think it is more efficient to let it where it is now - has less API calls. What do you say? Keep it as it is or proceed with the change?

Reply:
Makes sense, I think its fine to keep it here then. If you want to safe another API call you could just call NetworkCreate and ignore the ErrNetworkExists error.

Reply:
That's tempting. :)
The only downside is that it will create log entries with "Request Failed...".
On the other hand it will avoid a possible race condition - if 2 kube plays are applied at the same time and check the network existence at the same time, both will want to create a new network and one of them will fail - if we ignore the ErrNetworkExists error  that will not happen.